core/src/main/scala/org/apache/spark/scheduler/TaskSchedulerImpl.scala

```diff
@@ -172,7 +172,7 @@ private[spark] class TaskSchedulerImpl private[scheduler](
 
     if (!isLocal && conf.getBoolean("spark.speculation", false)) {
       logInfo("Starting speculative execution thread")
-      speculationScheduler.scheduleAtFixedRate(new Runnable {
+      speculationScheduler.scheduleWithFixedDelay(new Runnable {
```

Comment:
I think you are right, this is a good change.  Its somewhat related to other changes here, so I personally don't feel too strongly about needing to put it in its own pr.

I would say that if its in this pr, the change description should be updated to mention this as well.